core/templates/dev/head/pages/exploration_player/HintsAndSolutionManagerServiceSpec.js

```diff
@@ -99,6 +103,40 @@ describe('HintsAndSolutionManager service', function() {
     expect(hasms.isSolutionViewable()).toBe(true);
   });
 
+  it('should not continue to display hints after after a correct answer is' +
+     'submitted',
+    function() {
+      expect(hasms.isHintViewable(0)).toBe(false);
+      expect(hasms.isHintViewable(1)).toBe(false);
+      expect(hasms.isSolutionViewable()).toBe(false);
+
+      $timeout.flush();
+      // The first hint becomes viewable.
+      expect(hasms.isHintViewable(0)).toBe(true);
+      expect(hasms.isHintViewable(1)).toBe(false);
+      expect(hasms.isSolutionViewable()).toBe(false);
+
+      // The first hint is consumed, but a delay is needed for the second hint
+      // to be viewable.
+      expect(hasms.displayHint(0).getHtml()).toBe('one');
+      expect(hasms.isHintViewable(0)).toBe(true);
+      expect(hasms.isHintViewable(1)).toBe(false);
+      expect(hasms.isSolutionViewable()).toBe(false);
+
+      $rootScope.$broadcast(EVENT_NEW_CARD_AVAILABLE);
+      $timeout.flush();
+
+      // Because a correct answer was submitted, the next hint should not be
+      // available.
```

Comment:
Are there three hints? The format of the test implies there are only two.

Reply:
I think there are two

Reply:
Ah ok, I think I misread. Looks good!